### PR TITLE
APIGW: expand coverage for API Keys and Usage Plans

### PR DESCRIPTION
**Pull request by ArthurAkh**
## Motivation
We expand coverage for API Keys and Usage Plans.

## Changes
- Add tests for:
  - `get_api_keys` / `get_usage_plan_keys` nameQuery (prefix/postfix/infix, case-insensitive) and pagination (`limit` + `position`)
  - API Key edge cases (name/value/description lengths, patch validation for immutable fields and enabled type)
  - E2E: create API Key → create Usage Plan → link via UsagePlanKey → list & query keys
- Snapshot updates

### CODE REVIEW

File: tests/aws/services/apigateway/test_apigateway_extended.py [hunk 1]
```diff
@@ -272,7 +268,7 @@ def test_get_usage_plan_api_keys(self, aws_client, apigw_create_api_key, snapsho
         snapshot.match("create-api-key", create_api_key)
 
         create_api_key_2 = apigw_create_api_key(name=api_key_name_2)
-        snapshot.match("create-api-key-2", create_api_key)
+        snapshot.match("create-api-key-2", create_api_key_2)
```
Comment (bentsku) [inline tests/aws/services/apigateway/test_apigateway_extended.py:271]: great catch 🚀

File: tests/aws/services/apigateway/test_apigateway_extended.py [hunk 2]
```diff
@@ -332,3 +328,141 @@ def test_get_usage_plan_api_keys(self, aws_client, apigw_create_api_key, snapsho
...
+            aws_client.apigateway.update_api_key(
+                apiKey=api_key_id,
+                patchOperations=[{"op": "replace", "path": "/id", "value": "new-id"}],
+            )
+        snapshot.match("update-api-key-immutable-field", e.value.response)
+
+    @markers.aws.validated
+    def test_create_usage_plan_with_invalid_throttle(
+        self, aws_client, snapshot, cleanup_api_keys, cleanups
+    ):
```
Comment (bentsku) [inline tests/aws/services/apigateway/test_apigateway_extended.py]: question: naming of the test seems to indicate something should fail, but it doesn't, should the test be renamed?

File: localstack-core/localstack/services/apigateway/legacy/provider.py [hunk 3]
```diff
@@ -2392,6 +2402,9 @@ def get_api_keys(
         if not include_values:
             for api_key in api_keys:
                 api_key.pop("value")
+        if limit is not None:
+            if limit < 1 or limit > 500:
+                limit = None
```
Comment (bentsku) [inline localstack-core/localstack/services/apigateway/legacy/provider.py]: nit: I think the PR description mentions pagination, but there are no tests for it, so I'd suggest to just drop this new code and merge it like that, the newly added validation is already really good 👍

**Review by bentsku — COMMENTED** [review]
Nice work! Great added validation 🚀 

Just 2 comments: one test name looks like it should fail, so maybe this should be updated. The other one is the PR description mentioning pagination, I think we could remove it from the scope of this PR 👍 

Once this is addressed, we're good to merge! 👌

File: localstack-core/localstack/services/apigateway/legacy/provider.py (hunk 3 above)
Comment (bentsku) [reply localstack-core/localstack/services/apigateway/legacy/provider.py]: there was actually a test for the `-1` value 👍 added it back

**Review by bentsku — APPROVED** [review]
LGTM, I've resolved my comments, renamed the test only and this is now good to merge 👍 
I'll run a Pro pipeline just to be sure because we are un-validated tests there, but should be safe
